src/server/api/routers/company.ts

```diff
@@ -0,0 +1,77 @@
+import { z } from "zod";
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import { Industry } from "@prisma/client";
+
+export const companyRouter = createTRPCRouter({
+  getAll: publicProcedure.query(({ ctx }) => {
+    return ctx.db.company.findMany();
+  }),
+  getByName: publicProcedure
+    .input(
+      z.object({
+        name: z.string(),
+      }),
+    )
+    .query(({ ctx, input }) => {
+      return ctx.db.company.findFirst({
+        where: {
+          name: input.name,
+        },
+      });
+    }),
+  deleteByID: publicProcedure
+    .input(
+      z.object({
+        id: z.string(),
+      }),
+    )
+    .mutation(({ ctx, input }) => {
+      // TODO : should we delete all of the roles + reviews associated with a company?
```

Comment:
It might be helpful to look into [Referential actions](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/referential-actions) offered by Prisma. You can try adding the `onDelete: Cascade` action to relevant fields in our schema to delegate the deletion of related objects to Prisma when a deleting a company.